core/models.py

```diff
@@ -345,16 +361,48 @@ class Hackathon(MetaDataMixin):
         blank=True, null=True, help_text="List of items in the prize pool"
     )
 
-    image = models.ImageField(upload_to="hackathon_images")
+    fg_image = models.ImageField(upload_to="hackathon_images", null=True, blank=True)
+    bg_image = models.ImageField(upload_to="hackathon_images", null=True, blank=True)
     notes = models.TextField(blank=True, default="")
 
+    hybrid = models.CharField(
+        max_length=1,
+        choices=HYBRID_CHOICES,
+        default="I",
+        help_text="Location of the hackathon, I for in-person, V for virtual, H for hybrid",
+    )
+
+    # Boolean fields for whether the hackathon is specific to certain groups
+    is_web3 = models.BooleanField(default=False)  # Web 3 hackathons
+    is_diversity = models.BooleanField(
+        default=False
+    )  # Diversity hackathons for specific marginalized groups
+    is_restricted = models.BooleanField(
+        default=False
+    )  # Hackathons with restricrted enrollment (ex. only students of some university)
+    is_nonenglish = models.BooleanField(
+        default=False
+    )  # Hackathons which are not in English
+    is_over18 = models.BooleanField(
+        default=False
+    )  # Hackathons which are only for people over 18
+
+    freeze_data = models.BooleanField(
+        default=False,
+        help_text="Set to True to not update any details using scraped data. Use if you get accurate details directly from the hackathon organizers.",
+    )
+
+    custom_info = models.JSONField(
+        default=dict, null=True, blank=True
+    )  # Anything else that we might want to add in a structured format
+
     class Meta:
         ordering = ["start_date"]
 
         constraints = [
             models.CheckConstraint(  # ensure start date is before end date
-                check=models.Q(start_date__lt=models.F("end_date")),
```

Comment:
ehh... semantics but you generally don't want a start and end date (down to min iirc) to be at the same time.

Reply:
I mean the date is feb 12  2024 to feb 12 2024, what should I do ?